.github/workflows/quality-assurance.yml

```diff
@@ -1,56 +1,79 @@
-name: Ubuntu
+name: Quality Assurance
 
 on: [ push, pull_request ]
 
 jobs:
-  build:
-    env:
-      android-version: 26
-      JAVA_VERSION: ${{ matrix.java_version }}
+  generate-matrix:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: true
-      matrix:
-        java_version: [ 11, 17 ]
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Generate matrix
+        id: set-matrix
+        run: node .github/workflows/matrix.js
 
+  build-and-test:
+    needs: generate-matrix
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
     steps:
-      - uses: actions/checkout@v4
-      - name: "Set up JDK ${{ matrix.java_version }}"
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
```

Comment:
I think codecov needs the history to work correctly, remove here again (but keep the others)